client/components/tinymce/plugins/simple-payments/dialog/index.jsx

```diff
@@ -37,37 +37,45 @@ import {
 import { FEATURE_SIMPLE_PAYMENTS } from 'lib/plans/constants';
 import { hasFeature, getSitePlanSlug } from 'state/sites/plans/selectors';
 import UpgradeNudge from 'my-sites/upgrade-nudge';
+import TrackComponentView from 'lib/analytics/track-component-view';
+import analytics from 'lib/analytics';
 
 // Utility function for checking the state of the Payment Buttons list
 const isEmptyArray = a => Array.isArray( a ) && a.length === 0;
 
 // Selector to get the form values, insert the missing field, and convert it to
 // a custom post data structure ready to be passed to `wpcom` API.
-const productFormToCustomPost = state => {
+const getFormValues = state => {
 	const currencyCode = getCurrentUserCurrencyCode( state );
 	const formValues = getProductFormValues( state );
 
 	if ( currencyCode ) {
 		formValues.currency = currencyCode;
 	}
 
-	return productToCustomPost( formValues );
+	return formValues;
 };
 
 // Thunk action creator to create a new button
 const createPaymentButton = siteId => ( dispatch, getState ) => {
-	const productCustomPost = productFormToCustomPost( getState() );
+	const formValues = getFormValues( getState() );
+	const productCustomPost = productToCustomPost( formValues );
 
 	return wpcom.site( siteId ).addPost( productCustomPost ).then( newPost => {
 		const newProduct = customPostToProduct( newPost );
 		dispatch( receiveUpdateProduct( siteId, newProduct ) );
+		analytics.tracks.recordEvent( 'calypso_simple_payments_button_create', {
+			price: formValues.price,
+			currency: formValues.currency,
```

Comment:
You could use `newProduct.price` and `newProduct.currency` here and avoid renaming `productFormToCustomPost` to `getFormValues` and moving the `productToCustomPost` conversion out of it.

Makes the diff smaller, and the original code is more suitable to be eventually moved into the data layer.